101-data-share-share-storage-account/azuredeploy.json

```diff
@@ -0,0 +1,190 @@
+      "resources": [
+        {
+          "type": "blobServices/containers",
+          "apiVersion": "2019-06-01",
+          "name": "[concat('default/', variables('containerName'))]",
+          "dependsOn": [
+            "[variables('storageAccountName')]"
+          ]
+        }
+      ]
+    },
+    {
+      "type": "Microsoft.DataShare/accounts",
+      "apiVersion": "2019-11-01",
+      "name": "[variables('dataShareAccountName')]",
+      "location": "[parameters('location')]",
+      "identity": {
+        "type": "SystemAssigned"
+      },
+      "properties": {}
+    },
+    {
+      "type": "Microsoft.DataShare/accounts/shares",
+      "apiVersion": "2019-11-01",
+      "name": "[concat(variables('dataShareAccountName'), '/', variables('dataShareName'))]",
+      "dependsOn": [
+        "[resourceId('Microsoft.DataShare/accounts', variables('dataShareAccountName'))]"
+      ],
+      "properties": {
+        "shareKind": "CopyBased"
+      }
+    },
+    {
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2020-06-01",
+      "name": "[variables('roleAssignmentDeploymentName')]",
+      "resourceGroup": "[resourceGroup().name]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]",
+        "[resourceId('Microsoft.DataShare/accounts', variables('dataShareAccountName'))]"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": [
+            {
+              "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+              "apiVersion": "2020-04-01-preview",
+              "name": "[concat(variables('storageAccountName'), '/Microsoft.Authorization/', variables('roleAssignmentName'))]",
+              "properties": {
+                "roleDefinitionId": "[variables('storageBlobDataReaderRoleDefinitionId')]",
+                "principalId": "[reference(resourceId('Microsoft.DataShare/accounts', variables('dataShareAccountName')), '2019-11-01', 'Full').identity.principalId]"
+              }
+            }
+          ]
+        }
+      }
+    },
+    {
+      "type": "Microsoft.DataShare/accounts/shares/dataSets",
+      "apiVersion": "2019-11-01",
+      "name": "[concat(variables('dataShareAccountName'), '/', variables('dataShareName'),'/', variables('containerName'))]",
+      "dependsOn": [
+        "[resourceId('Microsoft.DataShare/accounts/shares', variables('dataShareAccountName'), variables('dataShareName'))]",
+        "[resourceId('Microsoft.DataShare/accounts', variables('dataShareAccountName'))]",
+        "[resourceId('Microsoft.Resources/deployments', variables('roleAssignmentDeploymentName'))]"
+      ],
+      "kind": "Container",
+      "properties": {
+        "subscriptionId": "[subscription().subscriptionId]",
```

Comment:
if these are always the current subscription/rg - then just remove them.  If not, they should be parameters

Reply:
Both subscriptionID and ResourceGroup are required.  I will make both as parameters.